cmd/notify-http.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+)
+
+type httpNotify struct {
+	Enable bool   `json:"enable"`
+	Addr   string `json:"address"`
+}
+
+func newHTTPNotify(accountID string) (*logrus.Logger, error) {
+	rNotify := serverConfig.GetHTTPNotifyByID(accountID)
+	fmt.Println(rNotify)
+
+	notifyLog := logrus.New()
+
+	notifyLog.Out = ioutil.Discard
+
+	// Set default JSON formatter.
+	notifyLog.Formatter = new(logrus.JSONFormatter)
+
+	notifyLog.Hooks.Add(rNotify)
+
+	// Success
+	return notifyLog, nil
+}
+
+// Fire is called when an event should be sent to the message broker.
+func (n httpNotify) Fire(entry *logrus.Entry) error {
+
+	// Fetch event type upon reflecting on its original type.
+	entryStr, ok := entry.Data["EventType"].(string)
+	if !ok {
+		return nil
+	}
+	fmt.Println("We got a HTTP event fired.")
+	fmt.Println(entryStr)
+
+	buf := bytes.NewBufferString(entryStr)
+	response, err := http.Post(n.Addr, "application/json", buf)
```

Comment:
@harshavardhana please checkout the timeouts I added to the Client. I'm also posting a JSON body now:

```
{"Records":[{"eventVersion":"2.0","eventSource":"aws:s3","awsRegion":"us-east-1","eventTime":"2017-01-09T09:01:09Z","eventName":"s3:ObjectCreated:Put","userIdentity":{"principalId":"minio"},"requestParameters":{"sourceIPAddress":"127.0.0.1:61837"},"responseElements":{},"s3":{"s3SchemaVersion":"1.0","configurationId":"Config","bucket":{"name":"test","ownerIdentity":{"principalId":"minio"},"arn":"arn:aws:s3:::test"},"object":{"key":"id_rsa.pub","size":398,"eTag":"2fe65071f85bbb4967f2af78d560861e","sequencer":"14980EE8B47A4D6D"}}}],"Key":"test/id_rsa.pub"}
```